src/ratio.jl

```diff
@@ -0,0 +1,31 @@
+# Base's Rational type is too slow because of its penchant for calling gcd and div.
+# So we roll our own.
```

Comment:
Presumably we can't make `Rational` itself behave this way, because if you _don't_ call `gcd` and `div`, then overflow seems likely. But you're right that perhaps we could use a layered approach.

CCing @StefanKarpinski, @simonbyrne, and  @JeffBezanson. Executive summary: Interpolations is the successor to Grid, with an aim to improve the API and performance. At the top of this issue I linked to a performance problem that stemmed from using `Rational`, where I got a 50x speed boost by rolling my own stripped-down version (called `Ratio`). We're attracted to use something like `Rational` for coefficients like `1//2`, rather than floating-point numbers like `0.5`, because it works out better in terms of promotion behavior for different `Number` types. I don't immediately see a good way of designing `Rational` to get both behaviors, but wondering if you folks have any thoughts about this.